beater/onboarding.go

```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/elastic/beats/libbeat/logp"
 )
 
-func notifyListening(config *Config, reporter reporter) {
+func notifyListening(config *Config, pubFct func(e beat.Event)) {
```

Comment:
only nitpicking since you need to push again to resolve merge conflicts, can remove the `e` named param here